airflow-core/src/airflow/utils/helpers.py

```diff
@@ -244,11 +244,6 @@ def render_template_to_string(template: jinja2.Template, context: Context) -> st
     return render_template(template, cast("MutableMapping[str, Any]", context), native=False)
 
 
-def render_template_as_native(template: jinja2.Template, context: Context) -> Any:
-    """Shorthand to ``render_template(native=True)`` with better typing support."""
-    return render_template(template, cast("MutableMapping[str, Any]", context), native=True)
```

Comment:
I would not expect users to use this, but cannot stop them from using it either, does it make sense to add deprecation for this?

Reply:
I think it's worth doing so. I believe I saw at least few cases where users used it outside of regular airflow template rendering happening automatically - because it is so convenient to use in - for example - your `@python` decorated task. I think especially for people who transition and switch between "classic" and "taskflow" operators, using JINJA for a number of those transformations is very natural and straightforward - they can even copy or import the same JINJA templates to be used in both - classic operators and decorated operators, so that might even be a good "final" solution.

And rendering as native objects is actually quite cool feature.

Reply:
Handled in 051725ce7e

Reply:
I handled it better now, @potiuk do you think this way of compat is fine for partial modules?